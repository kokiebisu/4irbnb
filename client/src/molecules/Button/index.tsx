import React from 'react';

// atoms
import { Button } from 'atoms';

// blocks
import SearchButton from 'molecules/Button/button.search';
import CookieButton from 'molecules/Button/button.cookie';
import GlobeButton from 'molecules/Button/button.globe';

export type ButtonProps = {
  type?: string;
  onPress: () => void;
  sm?: boolean;
  md?: boolean;
  lg?: boolean;
  inverse?: boolean;
  name?: string;
};

interface mapProps {
  [key: string]: JSX.Element;
}

/**
 * Use a button when you want to perform specific actions
 **/
export const CustomButton = ({ type, ...props }: ButtonProps) => {
  if (!type) {
    return <Button {...props} />;
  }
  const types: mapProps = {
    search: <SearchButton {...props} />,
    cookie: <CookieButton {...props} />,
    globe: <GlobeButton {...props} />,
  };
  return types[type];
};
