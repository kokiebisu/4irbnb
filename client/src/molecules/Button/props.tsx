export type Props = {
  type?: string;
  onPress: () => void;
  sm?: boolean;
  md?: boolean;
  lg?: boolean;
  inverse?: boolean;
  name?: string;
};
