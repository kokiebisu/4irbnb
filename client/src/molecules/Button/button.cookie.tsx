import React from 'react';
import { css } from 'styled-components';
import { Button, Text } from 'atoms';
import { colors, sizes } from 'styles';
import { CustomProps } from 'molecules/Button';
import theme from 'styled-theming';
import { darken } from 'polished';

export const cookieButtonColors = {
  primary: theme('mode', {
    light: darken(0.25, colors.gray),
  }),
};

export const cookieButtonStyles = {
  wrapper: css`
    padding: 14px 20px;
    font-size: 16px;
    letter-spacing: 1.1;
    font-weight: 300;
    white-space: nowrap;
    min-width: none;
    margin-top: 25px;

    @media ${sizes.lg} {
      min-width: 125px;
      width: 100%;
      margin-top: initial;
    }
  `,
  inverse: css`
    border: 1px solid ${cookieButtonColors.primary};
    background-color: transparent;
  `,
  normal: css`
    color: ${colors.white};
    background-color: ${cookieButtonColors.primary};
  `,
};

export const CookieButton: React.FC<CustomProps> = ({
  inverse,
  name,
  ...props
}) => {
  return (
    <Button
      styles={
        inverse
          ? [...cookieButtonStyles.inverse, ...cookieButtonStyles.wrapper]
          : [...cookieButtonStyles.normal, ...cookieButtonStyles.wrapper]
      }
      {...props}>
      <Text>{name}</Text>
    </Button>
  );
};
