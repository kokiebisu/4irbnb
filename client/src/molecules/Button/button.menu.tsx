import React from 'react';
import styled, { css } from 'styled-components';
import { Box, Icon, Button } from 'atoms';
import { colors } from 'styles';
import { CustomProps } from 'molecules/Button';
import theme from 'styled-theming';
import { lighten } from 'polished';

const menuButtonColors = {
  background: theme('mode', {
    light: colors.white,
  }),
  content: theme('mode', {
    light: colors.gray,
  }),
  border: theme('mode', {
    light: lighten(0.4, colors.gray),
  }),
};

const menuButtonStyles = {
  wrapper: css`
    background-color: ${menuButtonColors.background};
    border-radius: 48px;
    padding: 5px 6px 5px 12px;
    transition: 0.2s ease-in;
    border: 1px solid ${menuButtonColors.border};
    &:hover {
      box-shadow: rgba(0, 0, 0, 0.15) 0px 2px 8px;
    }
  `,
  flex: css`
    display: flex;
    align-items: center;
  `,
  icon: css`
    svg {
      width: 100%;
      height: 100%;
      path {
        fill: ${menuButtonColors.content};
      }
    }
  `,
  bars: css`
    margin-right: 12px;
    height: 15px;
    width: 17px;
  `,
  avatar: css`
    height: 32px;
    width: 32px;
  `,
};

export const MenuButton: React.FC<CustomProps> = ({ name, ...props }) => {
  return (
    <Button styles={menuButtonStyles.wrapper} {...props}>
      <Box styles={menuButtonStyles.flex}>
        <Icon
          styles={[...menuButtonStyles.bars, ...menuButtonStyles.icon]}
          name='Bars'
        />
        <Icon
          styles={[...menuButtonStyles.avatar, ...menuButtonStyles.icon]}
          name='Avatar'
        />
      </Box>
    </Button>
  );
};
