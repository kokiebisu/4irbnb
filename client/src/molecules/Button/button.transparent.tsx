import React from 'react';
import { css } from 'styled-components';
import { Button } from 'atoms';
import { colors } from 'styles';
import { CustomProps } from 'molecules/Button';
import theme from 'styled-theming';

export const transparentButtonColors = {
  stroke: theme('mode', {
    light: colors.white,
  }),
  hover: theme('mode', {
    light: 'rgba(190, 190, 190, 0.2)',
  }),
};

const transparentButtonStyles = {
  wrapper: css`
    padding: 15px;
    background-color: transparent;
    border-radius: 24px;
    cursor: pointer;
    &:focus {
      outline: none;
    }
    &:hover {
      background-color: ${transparentButtonColors.hover};
    }
  `,
};

export const globeButtonStyles = {
  flex: css`
    display: flex;
    align-items: center;
  `,
  globe: css`
    position: relative;
    top: 50%;
    margin-right: 6px;
    width: 16px;
    height: 16px;
  `,
  arrow: css`
    width: 10px;
  `,
};

export const hostButtonStyles = {};

export const TransparentButton: React.FC<CustomProps> = ({
  name,
  children,
  ...props
}) => {
  return (
    <Button styles={transparentButtonStyles.wrapper} {...props}>
      {children}
    </Button>
  );
};
