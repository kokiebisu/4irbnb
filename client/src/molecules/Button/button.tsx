import React from 'react';

// atoms
import { BaseButton } from 'atoms/Button';

// blocks
import SearchButton from 'molecules/Button/button.search';
import CookieButton from 'molecules/Button/button.cookie';
import GlobeButton from 'molecules/Button/button.globe';

export type ButtonProps = {
  type?: string;
  onPress: () => void;
  sm?: boolean;
  md?: boolean;
  lg?: boolean;
  inverse?: boolean;
  name?: string;
};

interface mapProps {
  [key: string]: JSX.Element;
}

/**
 * Use a button when you want to perform specific actions
 **/
export const Button = ({ type, ...props }: ButtonProps) => {
  if (!type) {
    return <BaseButton {...props} />;
  }
  const types: mapProps = {
    search: <SearchButton {...props} />,
    cookie: <CookieButton {...props} />,
    globe: <GlobeButton {...props} />,
  };
  return types[type];
};
