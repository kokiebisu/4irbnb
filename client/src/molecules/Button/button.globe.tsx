import React from 'react';
import { css } from 'styled-components';

// atoms
import { Icon } from 'atoms/Icon';
import { BaseButton } from 'atoms/Button';
import { Box } from 'atoms/Box';

import { Flex } from 'layout/flex';

// type
import { ButtonProps } from 'molecules/Button/button';
import theme from 'styled-theming';

import { colorpallete } from 'styles/colorpallete';

export default ({ name, ...props }: ButtonProps) => {
  return (
    <BaseButton styles={extend.wrapper} {...props}>
      <Flex alignCenter>
        <Icon styles={extend.icon.globe} name='Globe' />
        <Icon styles={extend.icon.arrow} name='BottomArrow' />
      </Flex>
    </BaseButton>
  );
};

const stroke = theme('mode', {
  light: colorpallete.white,
});

const extend = {
  wrapper: css`
    padding: 10px 14px;
    background-color: transparent;
    border-radius: 24px;
    cursor: pointer;
    &:focus {
      outline: none;
    }
    &:hover {
      background-color: rgba(190, 190, 190, 0.2);
    }
  `,
  icon: {
    globe: css`
      position: relative;
      top: 50%;
      margin-right: 6px;
      width: 16px;
      height: 16px;
      svg {
        width: 100%;
        path {
          fill: ${stroke};
        }
      }
    `,
    arrow: css`
      width: 10px;
      svg {
        width: 100%;
        polygon {
          fill: ${stroke};
        }
      }
    `,
  },
};
