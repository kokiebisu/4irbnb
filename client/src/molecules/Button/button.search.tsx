import React from 'react';
import { css } from 'styled-components';

// breakpoints
import { sizes } from '../../styles/breakpoints';

// atoms
import { Icon } from 'atoms/Icon';
import { BaseButton } from 'atoms/Button';
import { Text } from 'atoms/Text';
import { Box } from 'atoms/Box';

// type
import { ButtonProps } from 'molecules/Button/button';

export default ({ name, ...props }: ButtonProps) => {
  return (
    <BaseButton styles={extend.button} {...props}>
      <Box styles={extend.box}>
        <Icon styles={extend.icon} name='search' />
        <Text styles={extend.text}>{name}</Text>
      </Box>
    </BaseButton>
  );
};

const extend = {
  button: css`
    background: -webkit-linear-gradient(
      right,
      rgb(230, 30, 77) 0%,
      rgb(227, 28, 95) 50%,
      rgb(215, 4, 102) 100%
    );
    border: none;
    color: white;
  `,
  box: css`
    display: block;
    @media ${sizes.sm} {
      display: flex;
      align-items: center;
    }
  `,
  icon: css`
    position: relative;
    top: 1.5px;

    & svg {
      width: 14px;
      height: 14px;
      stroke: white;
      path {
        fill: white;
        stroke-width: 2;
      }
    }
  `,
  text: css`
    display: none;
    font-size: 15px;
    letter-spacing: 0.5px;
    font-family: Circular, -apple-system, BlinkMacSystemFont, Roboto,
      Helvetica Neue, sans-serif !important;
    font-weight: 400;
    padding-left: 7px;

    @media ${sizes.sm} {
      display: block;
    }
  `,
};
