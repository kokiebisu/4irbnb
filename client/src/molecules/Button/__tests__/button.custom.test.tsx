import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

// atoms
import { CustomButton } from 'molecules/Button';

describe('Cookie Button', () => {
  it('renders correctly', () => {
    const testingData = {
      children: 'test',
      action: () => console.log('test'),
    };

    const tree = renderer
      .create(
        <CustomButton
          type='cookie'
          onPress={testingData.action}
          {...testingData}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
