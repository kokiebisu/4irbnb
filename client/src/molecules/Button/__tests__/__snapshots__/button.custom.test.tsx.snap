// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cookie Button renders correctly 1`] = `
.c0 {
  font-size: 18px;
  border: none;
  border-radius: 8px;
  color: #ffffff;
  padding: 14px 20px;
  font-size: 16px;
  -webkit-letter-spacing: 1.1;
  -moz-letter-spacing: 1.1;
  -ms-letter-spacing: 1.1;
  letter-spacing: 1.1;
  font-weight: 300;
  white-space: nowrap;
  min-width: none;
  margin-top: 25px;
}

.c0:hover {
  cursor: pointer;
}

.c0:focus {
  outline: none;
}

.c1 {
  margin: 0;
  font-family: 'Airbnb-Cereal';
}

@media (min-width:992px) {
  .c0 {
    min-width: 125px;
    width: 100%;
    margin-top: initial;
  }
}

<button
  action={[Function]}
  className="c0"
  onClick={[Function]}
  style={Object {}}
>
  <p
    className="c1"
    style={Object {}}
  />
</button>
`;
